### PR TITLE
Reader pre-Refresh: thumbnails for video cards 

### PR DESCRIPTION
**Pull request by samouri**
In the reader, each embed uses iframes which degrade the user experience.  By showing a thumbnail and only loading the iframe when clicked, we can make the reader experience smoother

#8521 

Note: each embedType needs to be added separately because they each have their own way of storing thumbnails (most share a way of specifying autoplay)

### CODE REVIEW

**Comment by samouri** [comment]
Notes:

1. Do we usually have unit tests for this stuff? I'm thinking for the `getThumbnailUrl` and `getAutoplayingIframe` functions.
1. The regex to parse out `videoId` from youtube is particularly sketchy, is it too sketchy?

**Comment by samouri** [comment]
If we decide that this should be integrated into pre-refresh, then it can be merged after review/testing/fixing

Else, can be merged after:
-  [ ] uncommit the modified styles / `featured-asset.jsx`

File: client/reader/stream/_style.scss [hunk 1]
```diff
@@ -240,6 +240,10 @@
 		display: block;
 	}
 
+	.reader__post-featured-video-thumbnail {
+		width: 100% !important;
```
Comment (blowery) [inline client/reader/stream/_style.scss]: why the `!important`? we try to avoid it if we can.

**Comment by blowery** [comment]
http://calypso.localhost:3000/read/feeds/23559910 is breaking on the first post ( http://calypso.localhost:3000/read/feeds/23559910/posts/1185163064 )

**Comment by blowery** [comment]
http://calypso.localhost:3000/read/feeds/12098 / http://calypso.localhost:3000/read/feeds/12098/posts/1185158064 is busted, similar issue. looks like the regexp isn't grabbing the right id.

**Comment by blowery** [comment]
http://calypso.localhost:3000/read/feeds/23920982 / http://calypso.localhost:3000/read/feeds/23920982/posts/1185057541

The thumb disappears and the video never appears when you click on the play icon

File: client/lib/post-normalizer/rule-content-detect-embeds.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ export default function detectEmbeds( post, dom ) {
 			const match = iframe.src.match( regExp );
 			const videoId = match && ( match && match[ 2 ].length === 11 ) ? match[ 2 ] : false;
 
-			return videoId ? `https://img.youtube.com/vi/${ videoId }/0.jpg` : null;
+			return videoId ? `http://img.youtube.com/vi/${ videoId }/mqdefault.jpg` : null;
```
Comment (samouri) [inline client/lib/post-normalizer/rule-content-detect-embeds.js]: thanks for catching that! I'll add the _s_ back in

File: client/reader/stream/_style.scss (hunk 1 above)
Comment (samouri) [inline client/reader/stream/_style.scss]: I'm not sure I can avoid it.
Need to combat the `!important` over here: https://github.com/Automattic/wp-calypso/pull/8640/files#diff-5fc366969c3080c52fb50121347ef0e3R226

**Comment by samouri** [comment]
@blowery: turns out `;`s are only accepted in some cases as a parameter delimiter which was breaking playback.  By switching to `&` things are working again 😄

**Comment by samouri** [comment]
Found a bug with embed resizing logic - I'll open up a new issue for it.
The post: https://wordpress.com/read/feeds/23920982/posts/1185057541
in feed: https://wordpress.com/read/feeds/23920982
has an iframe with 0 height and 0 width. This was unnoticeable before (unless you looked at source), but now with thumbnails correctly showing, it is a problem

**Comment by samouri** [comment]
not implementing pre-refresh, refreshed version is here: #8704

**Pull request closed without merging**